airbyte-integrations/connectors/source-amazon-ads/source_amazon_ads/source.py

```diff
@@ -0,0 +1,114 @@
+    SponsoredProductCampaigns,
+    SponsoredProductKeywords,
+    SponsoredProductNegativeKeywords,
+    SponsoredProductsReportStream,
+    SponsoredProductTargetings,
+)
+
+TOKEN_URL = "https://api.amazon.com/auth/o2/token"
+
+
+# Source
+class SourceAmazonAds(AbstractSource):
+    def __init__(self):
+        super().__init__()
+        self.ctx = SourceContext()
+
+    def check_connection(self, logger: AirbyteLogger, config: Mapping[str, Any]) -> Tuple[bool, any]:
+        """
+        :param config:  the user-input config object conforming to the connector's spec.json
+        :param logger:  logger object
+        :return Tuple[bool, any]: (True, None) if the input config can be used to connect to the API successfully, (False, error) otherwise.
+        """
+        config = Spec(**config)
+        Profiles(config, authenticator=self._make_authenticator(config)).fill_context()
+        return True, None
+
+    def streams(self, config: Mapping[str, Any]) -> List[Stream]:
+        """
+        :param config: A Mapping of the user input configuration as defined in the connector spec.
+        """
+        config = Spec(**config)
+        auth = self._make_authenticator(config)
+        profiles_stream = Profiles(config, context=self.ctx, authenticator=auth)
+        profiles_stream.fill_context()
+        return [
+            profiles_stream,
+            SponsoredDisplayCampaigns(config, context=self.ctx, authenticator=auth),
+            SponsoredDisplayAdGroups(config, context=self.ctx, authenticator=auth),
+            SponsoredDisplayProductAds(config, context=self.ctx, authenticator=auth),
+            SponsoredDisplayTargetings(config, context=self.ctx, authenticator=auth),
+            DisplayReportStream(config, context=self.ctx, authenticator=auth),
+            SponsoredProductCampaigns(config, context=self.ctx, authenticator=auth),
+            SponsoredProductAdGroups(config, context=self.ctx, authenticator=auth),
+            SponsoredProductKeywords(config, context=self.ctx, authenticator=auth),
+            SponsoredProductNegativeKeywords(config, context=self.ctx, authenticator=auth),
+            SponsoredProductAds(config, context=self.ctx, authenticator=auth),
+            SponsoredProductTargetings(config, context=self.ctx, authenticator=auth),
+            SponsoredProductsReportStream(config, context=self.ctx, authenticator=auth),
+            SponsoredBrandsCampaigns(config, context=self.ctx, authenticator=auth),
+            SponsoredBrandsAdGroups(config, context=self.ctx, authenticator=auth),
+            SponsoredBrandsKeywords(config, context=self.ctx, authenticator=auth),
+            SponsoredBrandsReportStream(config, context=self.ctx, authenticator=auth),
+        ]
+
+    def spec(self, *args) -> ConnectorSpecification:
+        return ConnectorSpecification.parse_obj(Spec.schema())
+
+    @staticmethod
+    def _make_authenticator(config: Spec):
+        return Oauth2Authenticator(
```

Comment:
usually it is better to use keywords for anything having more than one arg

Reply:
done